components/push/ffi/src/lib.rs

```diff
@@ -68,16 +68,18 @@ pub extern "C" fn push_subscribe(
     handle: u64,
     channel_id: FfiStr<'_>,
     scope: FfiStr<'_>,
+    app_key: FfiStr<'_>,
     error: &mut ExternError,
 ) -> ByteBuffer {
     log::debug!("push_get_subscription");
     use push::msg_types::{KeyInfo, SubscriptionInfo, SubscriptionResponse};
     MANAGER.call_with_result_mut(error, handle, |mgr| -> Result<_> {
         let channel = channel_id.as_str();
         let scope_s = scope.as_str();
+        let app_key = app_key.as_opt_str();
```

Comment:
Will this actually work with the empty string you're passing?

Reply:
If I understand correctly (and I'm not 100% sure I do), `as_opt_str()` converts an optional kotlin arg into an `Option<&str>`. As per other uses in this file. The test I use were not passing an argument, so that should have resulted in `app_key` being set to `None`. My local testing appears to be old and not always reliable. I've added an appKey field to one of the test cases to ensure that nothing blows up.

Reply:
IIUC `as_opt_str` (definition below) would transform your empty string to a `Some(String(""))`:

https://github.com/mozilla/application-services/blob/06f7974549d2f445acb7f822a2a4cc27b41bb42a/components/support/ffi/src/ffistr.rs#L115-L128

Maybe I am missing another part that converts empty to null somewhere?

Reply:
I think in general we should avoid these `""` string equals `null` patterns since they bring confusion and use proper optionals/null pointers.

Reply:
Ah! fair enough. I misunderstood your original point. I'll adjust the protobuf.

Reply:
Yeah, "" isn't treated as None by as_opt_str. I think I'd be against changing that too, unless we were to do it by adding a new method or something, but IDK what it would be called.